x-pack/plugins/secops/public/components/formatted_duration/helpers.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+
+import { getEmptyValue } from '../empty_value';
+
+import * as i18n from './translations';
+
+/** one millisecond (as nanoseconds) */
+export const ONE_MILLISECOND_AS_NANOSECONDS = 1000000;
+
+export const ONE_SECOND = 1000;
+export const ONE_MINUTE = 60000;
+export const ONE_HOUR = 3600000;
+export const ONE_DAY = 86400000; // ms
+export const ONE_MONTH = 2592000000; // ms
+export const ONE_YEAR = 31536000000; // ms
+
+const milliseconds = (duration: moment.Duration): string =>
+  Number.isInteger(duration.milliseconds())
+    ? `${duration.milliseconds()}ms`
+    : `${duration.milliseconds().toFixed(6)}ms`; // nanosecond precision
+const seconds = (duration: moment.Duration): string =>
+  `${duration.seconds().toFixed()}s${
+    duration.milliseconds() > 0 ? ` ${milliseconds(duration)}` : ''
+  }`;
+const minutes = (duration: moment.Duration): string =>
+  `${duration.minutes()}m ${seconds(duration)}`;
+const hours = (duration: moment.Duration): string => `${duration.hours()}h ${minutes(duration)}`;
+const days = (duration: moment.Duration): string => `${duration.days()}d ${hours(duration)}`;
+const months = (duration: moment.Duration): string =>
+  `${duration.years() > 0 || duration.months() > 0 ? `${duration.months()}m ` : ''}${days(
+    duration
+  )}`;
+const years = (duration: moment.Duration): string =>
+  `${duration.years() > 0 ? `${duration.years()}y ` : ''}${months(duration)}`;
+
+export const getFormattedDurationString = (
+  maybeDurationNanoseconds: string | number | object | undefined | null
+): string => {
+  const totalNanoseconds = Number(maybeDurationNanoseconds);
+
+  if (maybeDurationNanoseconds == null) {
+    return getEmptyValue();
+  }
```

Comment:
I would move this up to line 45 so that you check for invariant conditions before calling into this function.

Also, weird stuff with Number is that you can have an empty string and it will turn into zero which looking at the unit tests I do not think you want.

Test case where that fails
```ts
    test('it echos back an empty string when the input is an empty string', () => {
      expect(getFormattedDurationString('')).toEqual('');
    });
```

I think you want to check and return for that case correctly as it is common.

Other cases such as `Number('0x11')`  which returns 17 is probably fine and I won't mention those.